### DISCUSSION TITLE
Help with recovering files on MacOS (osu!lazer)

### DISCUSSION THREAD

**Discussion by phucdtd** [ppy/osu#28536] [Q&A]
Whenever I've updated lazer on my Macbook, it has always prompted me to download a newer version (a .zip file), and then it can run "on top of the current version".

So far, I've done that, but when I double-clicked the zip file, it extract the osu!lazer application package to the same folder, so what I ended up doing is I copy it to the Application folder and replace the old one. However, sometimes, there existed another copy of the application package elsewhere, so I've then had a few copies of the osu!lazer app on my machine, on different versions.

Typically, this does not bother me since I only keep the newest version (or its shortcut) in the Application folder, but today I decided to clean it up. The reason is because when I open an .osr file, it defaulted to a 2022 version instead of 2024, and I was having trouble changing the default application to open the files, and don't want to drag it to the game every single time.

However, when I deleted the old versions, and relaunched the game, practically all of my data (skins, beatmaps, replays, etc.) were gone. I panicked and restored everything from the Trash folder (to how it was before), but when I relaunched nothing came back.

I've tried to look at the log at the Application Support folder, but there's no folder named "osu" in there. I've tried to do a `sudo find / -name client.realm` and managed to locate the db files at /Users/[name]/.local/share/osu/, and it shows as
<img width="669" alt="image" src="https://github.com/ppy/osu/assets/20723661/bb7b1863-bb4d-4518-816b-8566778be07f">

I've also found the logs there, and I believe this one is the logs upon my first time booting up the game to find my files gone, and the one right before:

[Before](https://github.com/user-attachments/files/15910955/1718864922.runtime.log)
[After](https://github.com/user-attachments/files/15910973/1718867884.runtime.log)

I'd appreciate any follow-up on what I can do to restore the data, and what would be my best move on updating moving forward. Thank you!

**Comment by peppy** [comment]
Please fine the `client_***_newer_version_before_recovery` that has the largest filesize and rename it to `client.realm` (moving `client.realm` out of the way first).

**Reply by phucdtd** [reply]
All of them are all equal in size at 66KB, except for `client_newer_version.v22.backup.realm`, which is 2.2MB in size.
I've made a folder for backups inside the main folder, and try each of them in turn.

The 66KB files are all empty databases with nothing. I tried the 2.2MB one and it turns out to be my db from quite a while ago. I've no idea how it survived, but before deleting the versions I have opened them up in some order, maybe that might have replaced my most recent db with an old db somewhere?

The 2.2MB db is still better than nothing, but the most recent score on there is from around 1 year ago. I think I'll still have to re-download everything, I'm just a bit sad seeing my local scores gone because I do tend to play offline from time to time when there's no connection.

Thanks for the help peppy, and I'd be glad to know if there's anything I can still try. I appreciate it!

**Reply by peppy** [reply]
hmm, try doing a system-wide search (`client*.realm`). maybe there's a newer version in a different location.

**Reply by phucdtd** [reply]
I just did a `sudo find -E / -regex '.*realm'` and all it managed to find is the directory shown above, containing all the files i've already tried tucked inside a backup folder.

**Reply by peppy** [reply]
not too sure then sorry :(
